dra-evolution/pkg/api/claim_types.go

```diff
@@ -1,379 +1,467 @@
 
-	// Requests allows the user to specify the minimum resource
-	// requirements that must be satisfied across all devices allocated for
-	// this claim.
+	// ReservedFor indicates which entities are currently allowed to use
+	// the claim. A Pod which references a ResourceClaim which is not
+	// reserved for that Pod will not be started.
 	//
-	// All drivers can the support "devices", resource, which represents
-	// the minimum number of devices to allocate.
+	// There can be at most 32 such reservations. This may get increased in
+	// the future, but not reduced.
 	//
-	// Other resource types are driver-specific. The values in `requests`
-	// represent the aggregate across all devices, not the per-device
-	// values.
+	// +listType=map
+	// +listMapKey=uid
+	// +patchStrategy=merge
+	// +patchMergeKey=uid
+	// +optional
+	ReservedFor []ResourceClaimConsumerReference `json:"reservedFor,omitempty" protobuf:"bytes,3,opt,name=reservedFor" patchStrategy:"merge" patchMergeKey:"uid"`
+}
+
+// AllocationResult contains attributes of an allocated resource.
+type AllocationResult struct {
+	// DriverData contains the state associated with an allocation that
+	// should be maintained throughout the lifetime of a claim. Each
+	// entry contains data that should be passed to a specific kubelet
+	// plugin once the claim lands on a node.
 	//
-	// If not set, requests.devices will default to 1.
+	// Setting this field is optional. It has a maximum size of 32 entries.
+	// If empty, nothing was allocated for the claim and kubelet does not
+	// need to prepare anything for it.
 	//
+	// +listType=atomic
 	// +optional
-	Requests map[string]resource.Quantity `json:"requests,omitempty"`
+	DriverData []DriverData `json:"driverData,omitempty" protobuf:"bytes,1,opt,name=driverData"`
 
-	// MatchAttributes allows specifying a constraint within a set of
-	// chosen devices, without having to explicitly specify the value of
-	// the constraint.  For example, this allows constraints like "all
-	// devices must be the same model", without having to specify the exact
-	// model. We may be able to use this for some basic topology
-	// constraints too, by representing the topology as device attributes.
-	//
-	// Currently, these are just strings. However, we could make them
-	// structs, and include required vs preferred matches. Required matches
-	// would fail if not met, where as preferred would lower the score if
-	// not met. We could even allow low/medium/high priority and adjust the
-	// score differently for each.
+	// This field defines where Pods can be scheduled which reference
+	// an allocated claim.
 	//
+	// Setting this field is optional. If null, the resource is available
+	// everywhere.
 	// +optional
-	// +listType=atomic
-	MatchAttributes []string `json:"matchAttributes,omitempty"`
+	AvailableOnNodes *v1.NodeSelector `json:"availableOnNodes,omitempty" protobuf:"bytes,2,opt,name=availableOnNodes"`
```

Comment:
So, this is a node selector because of the future possibility of network-reachable allocated claims?

Reply:
Yes.

I don't know whether it makes sense to start with just a node name and then add a selector later. Seems more complex to me.

Reply:
See https://github.com/kubernetes-sigs/wg-device-management/pull/14#discussion_r1625130497 - it's now `NodeName string`.